### DISCUSSION TITLE
How to use before_post_render filter

### DISCUSSION THREAD

**Discussion by lsy0xcc** [hexojs/hexo#4879] [Question / Answer] [answered]
I put 
```
console.log("script works");

hexo.extend.filter.register("before_post_render", function (data) {
  console.log(data);
  console.log("before_post_render");
  data.title = "before_post_render works"
  return data;
});

hexo.extend.filter.register('before_generate', function(){
  console.log("before_generate");
}); 
```

in hexoroot\script\test.js.

But the console only shows 
```
INFO  Validating config
script works
INFO  Start processing
before_generate
INFO  Hexo is running at http://localhost:4000/ . Press Ctrl+C to stop.
```
The title of posts not changes and seems before_post_render not worked
Can anyone tell me how to use before_post_render filter please?

**Comment by SukkaW** [comment]
`before_post_render` will only be invoked when a post page is about to be rendered. You can see the title change once visiting a post page.

However, if you are going to change the title of the post, `before_post_render` is a bit too late.

**Reply by lsy0xcc** [reply]
I forgot to clean the project, now it works. Thanks!

**Comment by yoshinorin** [comment, marked as answer]
@lisongyuan99 
I assume it is caused by the cache. Hexo has the cache (`db.json`). 
The `before_post_render` is perhaps trigger once after change a post.

Please try `hexo clean` if you want to trigger `before_post_render` without change a post.  (Also, you can delete it manually)

> https://hexo.io/docs/commands#clean

**Reply by lsy0xcc** [reply]
It works! thanks!!!
